xmrstak/backend/amd/amd_gpu/gpu.cpp

```diff
@@ -449,68 +449,85 @@ uint32_t getNumPlatforms()
 std::vector<GpuContext> getAMDDevices(int index)
 {
 	std::vector<GpuContext> ctxVec;
-	cl_platform_id * platforms = NULL;
+	std::vector<cl_platform_id> platforms;
+	std::vector<cl_device_id> device_list;
+
 	cl_int clStatus;
 	cl_uint num_devices;
-	cl_device_id *device_list = NULL;
-
 	uint32_t numPlatforms = getNumPlatforms();
 
-	if(numPlatforms)
+	if(numPlatforms == 0)
+		return ctxVec;
+
+	platforms.resize(numPlatforms);
+	if((clStatus = clGetPlatformIDs(numPlatforms, platforms.data(), NULL)) != CL_SUCCESS)
+	{
+		printer::inst()->print_msg(L1,"WARNING: %s when calling clGetPlatformIDs for platform information.", err_to_str(clStatus));
+		return ctxVec;
+	}
+
+	if((clStatus = clGetDeviceIDs( platforms[index], CL_DEVICE_TYPE_GPU, 0, NULL, &num_devices)) != CL_SUCCESS)
+	{
+		printer::inst()->print_msg(L1,"WARNING: %s when calling clGetDeviceIDs for of devices.", err_to_str(clStatus));
+		return ctxVec;
+	}
+
+	device_list.resize(num_devices);
+	if((clStatus = clGetDeviceIDs( platforms[index], CL_DEVICE_TYPE_GPU, num_devices, device_list.data(), NULL)) != CL_SUCCESS)
+	{
+		printer::inst()->print_msg(L1,"WARNING: %s when calling clGetDeviceIDs for device information.", err_to_str(clStatus));
+		return ctxVec;
+	}
+
+	for (int k = 0; k < num_devices; k++)
```

Comment:
nit: "k < num_devices" is a signed-int vs. unsigned-int comparison and may cause compiler warnings

Reply:
You are correct. @psychocrypt If you ever write `for(int ...` again I shall spank you ^^